lib/faktory_worker/worker_supervisor.ex

```diff
@@ -0,0 +1,38 @@
+defmodule FaktoryWorker.WorkerSupervisor do
+  @moduledoc false
+
+  use Supervisor
+
+  alias FaktoryWorker.Random
+
+  @spec start_link(opts :: keyword()) :: Supervisor.on_start()
+  def start_link(opts) do
+    name = Keyword.get(opts, :name)
+    Supervisor.start_link(__MODULE__, opts, name: :"#{name}_worker_supervisor")
+  end
+
+  @impl true
+  def init(opts) do
+    children =
+      opts
+      |> Keyword.get(:workers, [])
+      |> Enum.map(&map_worker/1)
+      |> List.flatten()
+
+    Supervisor.init(children, strategy: :one_for_one)
+  end
+
+  defp map_worker(worker_module) do
+    worker_opts = worker_module.worker_config()
+    concurrency = Keyword.get(worker_opts, :concurrency, 1)
+
+    Enum.reduce(1..concurrency, [], fn _, acc ->
+      worker_id = Random.worker_id()
+      worker_name = :"#{worker_module}_#{worker_id}"
```

Comment:
`worker_id` is currently also being generated in the connection, I guess the value here will be passed through in the next bit of work?

Reply:
Yep that's correct. This way we can easily tie workers in faktory directly to the worker servers in our supervision tree. The worker id also shows up in observer so it's really easy to find the worker and inspect its state.